gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/PageMapper.java

```diff
@@ -169,4 +173,142 @@ default JsonNode deserializeJsonConfiguration(Object configuration) throws JsonP
+     * This is a simple and reliable approach that doesn't require custom annotations.
+     */
+    @Named("deserializeConfigurationWithSensitiveDataMasking")
+    default Object deserializeConfigurationWithSensitiveDataMasking(String configuration) {
+        if (Objects.isNull(configuration)) {
+            return null;
+        }
+
+        try {
+            // Try to parse as LinkedHashMap first
+            Object config = mapper.readValue(configuration, LinkedHashMap.class);
+
+            // If we have a fetcher configuration, we need to mask sensitive fields
+            if (config instanceof LinkedHashMap) {
+                @SuppressWarnings("unchecked")
+                LinkedHashMap<String, Object> configMap = (LinkedHashMap<String, Object>) config;
+
+                // Apply sensitive data masking
+                maskSensitiveFieldsInMap(configMap);
+            }
+
+            return config;
+        } catch (JsonProcessingException jse) {
+            logger.debug(jse.getMessage());
+        }
+
+        try {
+            return mapper.readValue(configuration, List.class);
+        } catch (JsonProcessingException jse) {
+            logger.debug(jse.getMessage());
+        }
+
+        return configuration;
+    }
+
+    /**
+     * Get the list of sensitive field names that should be masked.
+     * This centralizes the sensitive field definitions to avoid duplication.
+     *
+     * @return array of sensitive field names
+     */
+    private String[] getSensitiveFieldNames() {
+        return new String[] {
+            "privateToken", // GitLab fetcher
+            "password", // HTTP, Git fetchers
+            "secret", // Various fetchers
+            "token", // Various fetchers
+            "apiKey", // API fetchers
+            "accessToken", // OAuth fetchers
+            "clientSecret", // OAuth fetchers
+            "personalAccessToken", // GitHub, GitLab
+            "authToken", // Generic auth tokens
+        };
+    }
+
+    /**
+     * Mask sensitive fields in a configuration map.
+     * This method checks for known sensitive field names and replaces them with "********".
+     */
+    default void maskSensitiveFieldsInMap(LinkedHashMap<String, Object> configMap) {
+        for (String fieldName : getSensitiveFieldNames()) {
+            if (configMap.containsKey(fieldName)) {
+                configMap.put(fieldName, SENSITIVE_DATA_REPLACEMENT);
+            }
+        }
+    }
+
+    /**
+     * Parse configuration JSON string to LinkedHashMap.
+     *
+     * @param configuration the configuration JSON string
+     * @return parsed configuration map or null if parsing fails
+     */
+    private LinkedHashMap<String, Object> parseConfigurationToMap(String configuration) {
+        if (Objects.isNull(configuration)) {
+            return null;
+        }
+
+        try {
+            Object config = mapper.readValue(configuration, LinkedHashMap.class);
```

Comment:
`readValue` is Generic so it directly maps to a LinkedHashMap.
Therefore the next check is useless 

This should work


```suggestion
            LinkedHashMap<String, Object> config = mapper.readValue(configuration, LinkedHashMap.class);
```

Reply:
Done